src/components/structures/TimelinePanel.tsx

```diff
@@ -806,23 +806,34 @@ class TimelinePanel extends React.Component<IProps, IState> {
         // Can be null for the notification timeline, etc.
         if (!this.props.timelineSet.room) return;
 
+        if (ev.getRoomId() !== this.props.timelineSet.room.roomId) return;
+
+        if (!this.state.events.includes(ev)) return;
+
+        this.recheckFirstVisibleEventIndex();
+
         // Need to update as we don't display event tiles for events that
         // haven't yet been decrypted. The event will have just been updated
         // in place so we just need to re-render.
         // TODO: We should restrict this to only events in our timeline,
         // but possibly the event tile itself should just update when this
         // happens to save us re-rendering the whole timeline.
-        if (ev.getRoomId() === this.props.timelineSet.room.roomId) {
-            this.buildCallEventGroupers(this.state.events);
-            this.forceUpdate();
-        }
+        this.buildCallEventGroupers(this.state.events);
+        this.forceUpdate();
     };
```

Comment:
I'm curious how this fixes https://github.com/vector-im/element-web/issues/16983

What stopped it from working before?

What changes were made to make history visible now?

Reply:
We weren't re-checking `checkForPreJoinUISI` when we decrypted events, there's a comment in `checkForPreJoinUISI` which says it treats on-going decryptions the same as failed decryptions.

Reply:
This specific bit doesn't do it; the `recheck` method does the trick - now when an event is decrypted we check if we can show older events, we didn't do that before

Reply:
Beat me to it (again today :D)